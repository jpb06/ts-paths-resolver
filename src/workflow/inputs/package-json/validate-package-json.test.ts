import { FileSystem } from '@effect/platform/FileSystem';
import { Effect, Layer, pipe } from 'effect';
import { runPromise } from 'effect-errors';
import { describe, expect, it } from 'vitest';

import { packageJsonMockData } from '@tests/mock-data';

import { InputError } from '../errors/input.error.js';

describe('validatePackageJson function', () => {
  it('should fail if file does not exist', async () => {
    const path = './package.json';

    const TestFileSystemlayer = Layer.succeed(
      FileSystem,
      FileSystem.of({
        exists: () => Effect.succeed(false),
      } as unknown as FileSystem),
    );

    const { validatePackageJson } = await import('./validate-package-json.js');

    const result = await Effect.runPromise(
      pipe(
        validatePackageJson(path),
        Effect.flip,
        Effect.provide(TestFileSystemlayer),
      ),
    );

    expect(result).toBeInstanceOf(InputError);
    expect((result as InputError).cause).toBe(
      `package.json file not found at ${path}`,
    );
  });

  it('should extract data from package.json', async () => {
    const path = './package.json';

    const TestFileSystemlayer = Layer.succeed(
      FileSystem,
      FileSystem.of({
        exists: () => Effect.succeed(true),
        readFileString: () => Effect.succeed(packageJsonMockData),
      } as unknown as FileSystem),
    );

    const { validatePackageJson } = await import('./validate-package-json.js');

    const result = await runPromise(
      pipe(validatePackageJson(path), Effect.provide(TestFileSystemlayer)),
    );

    expect(result).toMatchSnapshot();
  });
});
