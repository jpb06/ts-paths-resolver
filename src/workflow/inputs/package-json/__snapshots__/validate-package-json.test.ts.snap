// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`validatePackageJson function > should extract data from package.json 1`] = `
{
  "cjsEntryPoint": "./cjs/index.js",
  "dtsEntryPoint": "./dts/index.d.ts",
  "esmEntryPoint": "./esm/index.js",
}
`;
